src/odoc/bin/main.ml

```diff
@@ -712,6 +712,64 @@ module Odoc_error = struct
       ~doc:"Print errors that occurred while an .odoc file was generated."
 end
 
+module Odoc_uri : sig
+  val cmd : unit Term.t
+
+  val info : Term.info
+end = struct
+  let resolve directories reference =
+    let resolver =
+      Resolver.create ~important_digests:false ~directories ~open_modules:[]
+    in
+    let reference =
+      let open Odoc_model in
+      let warnings_options =
+        { Error.warn_error = true; print_warnings = true }
+      in
+      Semantics.parse_reference reference
```

Comment:
This gives bad locations for errors:
```
  ERROR: File "", line 0, characters 1-1:
  Identifier in reference should not be empty.
```

I think `Reference.parse location reference` should be used instead.

`Semantics.parse_reference` can be removed entirely, given that it's job is to actively harm. Let the wrong locations in the other call site for now.

Reply:
Well, that is a  good point, just that a reference passed on the command line has no location unlike compilation units and pages. As a consequence, we parsed the reference with a dummy location. In the same sense, I might be wrong and I am looking forward to a way forward :).

Reply:
That's right, the location type doesn't even allow location without a filename.
There's already a dummy location with `_none_` as filename, generated by `Location_.span []`. Perhaps `Location_.pp` can be improved to detect this dummy filename and adapt the error message ?

Reply:
That’s right, thank you!

Reply:
So we went for the following scenario:
- Filename is empty, in this case we do not display the error location at all: indeed, I don't think cmdliner allows to get the location of its arguments in terms of character position.
- Filename is "dummy", in this case we display the location: sometimes we do not have a file but we do have a character position/length.

That's certainly not a perfect approach, it does not uses ocaml types well, but we can still update it either here or on its own PR!

Reply:
Works for me :) The empty filename check is enough for now. The character position inside the string is the right value to show, certainly not the character position within the original command line. But if we want better we'd need to change the location type, let's do that later.